### PR TITLE
fix(nc-gui): allow expand when is locked

### LINKED ISSUE #3517: Expanded row view is not available for a locked view
## Steps : 
- Lock a grid view.
- Observe the expand row option is not available.

## Problem
- Expanded row view is not available after locking the grid view.

## Expected behaviour
- Expanded row view should be available.

### PR DESCRIPTION
**Pull request by mateonunez**
## Change Summary

removing `isLocked` check in `Smartsheet/Grid.vue` element

ref: #3517

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [x] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

This Pull Request may create conflicts with #4141.

### CODE REVIEW

**Comment by mateonunez** [comment]
I've converted this PR into a draft, need to run the E2E tests locally and check for behaviors in `rest-operation` tests.

**Comment by wingkwong** [comment]
Please rebase to include the fix of LTAR issues in CI/CD tests.

File: packages/nc-gui/components/smartsheet/Grid.vue [hunk 1]
```diff
@@ -566,7 +566,7 @@ watch(
                       <span class="flex-1" />
 
                       <div
-                        v-if="(!readOnly || hasRole('commenter', true) || hasRole('viewer', true)) && !isLocked"
+                        v-if="(!readOnly || hasRole('commenter', true) || hasRole('viewer', true))"
```
Comment (wingkwong) [inline packages/nc-gui/components/smartsheet/Grid.vue:569]: parentheses are not necessary. 

```
v-if="!readOnly || hasRole('commenter', true) || hasRole('viewer', true)"
```

**Review by wingkwong — CHANGES_REQUESTED** [review]

File: packages/nc-gui/components/smartsheet/Grid.vue (hunk 1 above)
Comment (mateonunez) [reply packages/nc-gui/components/smartsheet/Grid.vue]: Oh, I forgot to lint, thanks for the review! Fixed with a new commit

**Review by wingkwong — APPROVED** [review]

**Comment by wingkwong** [comment]
@dstala please help verify the cypress change

**Review by dstala — APPROVED** [review]

**Pull request merged**